src/coreclr/tools/dotnet-pgo/Program.cs

```diff
@@ -378,7 +378,17 @@ static int InnerMergeMain(CommandLineOptions commandLineOptions)
                     ProfileData.MergeProfileData(ref partialNgen, mergedProfileData, MIbcProfileParser.ParseMIbcFile(tsc, peReader, assemblyNamesInBubble, onlyDefinedInAssembly: null));
                 }
 
-                return MibcEmitter.GenerateMibcFile(tsc, commandLineOptions.OutputFileName, mergedProfileData.Values, commandLineOptions.ValidateOutputFile, commandLineOptions.Uncompressed);
+                int result = MibcEmitter.GenerateMibcFile(tsc, commandLineOptions.OutputFileName, mergedProfileData.Values, commandLineOptions.ValidateOutputFile, commandLineOptions.Uncompressed);
+                if (result == 0 && commandLineOptions.InheritTimestamp)
+                {
+                    static DateTime Latest(IEnumerable<DateTime> datetimes)
+                        => datetimes.Aggregate((cur, dt) => dt > cur ? dt : cur);
+
+                    commandLineOptions.OutputFileName.CreationTimeUtc = Latest(commandLineOptions.InputFilesToMerge.Select(fi => fi.CreationTimeUtc));
+                    commandLineOptions.OutputFileName.LastWriteTimeUtc = Latest(commandLineOptions.InputFilesToMerge.Select(fi => fi.LastWriteTimeUtc));
+                }
```

Comment:
You could do it in one chain 🙂:
```
commandLineOption.InputFilesToMerge
    .Select(fi => fi.CreationTimeUtc)
    .Combine(commandLineOptions.InputFilesToMerge.Select(fi => fi.LastWriteTimeUtc))
    .OrderByDescending(d => d).FirstOrDefault();
```

Reply:
I'm setting `CreationTimeUtc` and `LastWriteTimeUtc` separately here, so I would need two chains (or set both to the same, I guess that would work too). Actually, I see now that I can just use `.Max()` so will change it to that.